contracts/Core/StakeManager.sol

```diff
@@ -267,6 +266,9 @@ contract StakeManager is Initializable, ACL, StakeStorage, StateManager, Pause,
         uint256 sAmount = _convertRZRtoSRZR(lockedAmount, staker.stake, sToken.totalSupply());
 
         //Updating Staker Stake
+        if (staker.stake < parameters.minStake()) {
```

Comment:
Hey, why here this min stake condition is added, why it needs to be lesser than minStake ?

Reply:
Reason this was done because if the staker's stake is greater than minStake, then it means the staker can still vote in the network and hence, there is no need for his activity to be updated when he resets lock otherwise he may get away with the inactivity penalties when he commits. If the stake is less than minStake, the staker wont be able to vote in the network and therefore, it is important that his activity is updated when he resets lock.

Reply:
okay cool got it 
but I think its very extreme edge case.
without it the outcome of it , is staker who unstaked more then (stake-minStake), and does resetLock getting penalised for both resetLock and inactivity.

Reply:
CC : @hrishikeshio @SkandaBhat

Reply:
This creates a vulnerability where staker goes inactive for long time and unstakes when everyone else has unstaked and his stake goes below minstake. They will avoid inactivity penalty

Reply:
I think we should remove this.
If staker's stake is going below minstake then he can unstake to avoid penalty.
if he is not unstaking that means he is offline and he should be penalized.

Reply:
removed https://github.com/razor-network/contracts/pull/443/commits/74636206e264a357c578eaff329f19b6d2a137e8